### PR TITLE
Token stream wrapper

### PR DESCRIPTION
**Pull request by sultur**
Idea for a TokenStream wrapper class which contains an iterator and a small deque used for lookahead.

### CODE REVIEW

File: src/tokenizer/tokenizer.py [hunk 1]
```diff
@@ -843,6 +845,41 @@ def Split_Sentence(t: Optional[Tok] = None) -> Tok:
...
+    def __getitem__(self, i):
+        if 0 <= i < self.__max_lookahead:
+            l = len(self.__lookahead)
+            try:
+                while l <= i:
+                    # Extend deque to lookahead
+                    self.__lookahead.append(next(self.__it))
+                    l += 1
+                return self.__lookahead[i]
+            except StopIteration:
```
Comment (vthorsteinsson) [inline src/tokenizer/tokenizer.py:878]: Be careful that the StopIteration is not lost (it probably isn't, as next() probably raises StopIteration every time it's called on an empty iterator, but it needs to be confirmed).

File: src/tokenizer/tokenizer.py [hunk 2]
```diff
@@ -843,6 +845,41 @@ def Split_Sentence(t: Optional[Tok] = None) -> Tok:
         return t
 
 
+class TokenStream:
+    """
+    Wrapper for token iterator allowing lookahead.
+    """
+
+    def __init__(self, token_it, *, lookahead_size=2):
```
Comment (vthorsteinsson) [inline src/tokenizer/tokenizer.py]: Add type annotations

**Review by vthorsteinsson — COMMENTED** [review]
This is looking really good!

File: src/tokenizer/tokenizer.py (hunk 1 above)
Comment (sultur) [reply src/tokenizer/tokenizer.py:878]: Yes, `next` on an empty iterator raises StopIteration

File: src/tokenizer/tokenizer.py (hunk 2 above)
Comment (sultur) [reply src/tokenizer/tokenizer.py]: Added!

**Pull request merged**